ndl/sharadar/sharadar.go

```diff
@@ -190,3 +199,90 @@ func (d *Dataset) BulkDownloadPrices(ctx context.Context, table TableName) error
 	}
 	return nil
 }
+
+// ComputeActions from RawActions and Prices of the Dataset. This method must be
+// called after all the data is downloaded. It combines multiple actions between
+// price points, creating at most one action per price point.
+func (d *Dataset) ComputeActions(ctx context.Context) {
+	// For each ticker, step through the prices and actions in sync by date, and
+	// generate the appropriate DB actions.
+	for ticker, prices := range d.Prices {
+		if len(prices) == 0 {
+			continue
+		}
+		rawActions := d.RawActions[ticker]
+
+		actions := []db.ActionRow{}
+		var prevPrice float32 // split-adjusted previous close
+		ai := 0               // action index
+		for i, price := range prices {
+			action := db.ActionRow{
```

Comment:
Skip the remaining prices when done with all actions, except `i==0`, since we may need to inject a listed action.